viamonvif/discovery.go

```diff
@@ -109,6 +110,17 @@ type CameraInfo struct {
 	HardwareID      string   `json:"hardware_id"`
 }
 
+// regex to remove non alpha numerics.
+var reg = regexp.MustCompile("[^a-zA-Z0-9]+")
+
+// Name creates generates a name for the camera based on discovered information about the camera.
+func (cam *CameraInfo) Name(channel int) string {
+	stripManufacturer := reg.ReplaceAllString(cam.Manufacturer, "")
+	stripModel := reg.ReplaceAllString(cam.Model, "")
+	stripSerial := reg.ReplaceAllString(cam.SerialNumber, "")
+	return fmt.Sprintf("%s-%s-%s-channel%v", stripManufacturer, stripModel, stripSerial, channel)
```

Comment:
Wondering if we should change "channel" to "uri" or "url" to avoid conflating the index here with the actual route that may or may not include a channel number.